### PR TITLE
DevContainer QoL change

### PR DESCRIPTION
**Pull request by jacobsee**
We pass several ports from this devcontainer compose stack (NATS, Hydra, CRDB...) through to the host using the `ports` option directly in the `docker-compose.yml` file... but we have been omitting the port used by the governor-api app itself from the compose file and instead forwarded it using the `devcontainer.json` `forwardPorts` definition.

This works seamlessly in VSCode, however, `forwardPorts` is unsupported by the devcontainer CLI for anyone using another editor/IDE. The main difference between forwarding a port in the compose file vs. using `forwardPorts` is that `forwardPorts` is necessary for services that only listen on 127.0.0.1 - however, this service listens on 0.0.0.0, thus there is no functional difference here, so proposing this change for QoL/compatibility reasons.

### CODE REVIEW

File: .devcontainer/docker-compose.yml [hunk 1]
```diff
@@ -20,6 +19,8 @@ services:
     depends_on:
       - nats
     command: sleep infinity
+    ports:
+      - "3001:3001"
```
Comment (jnschaeffer) [inline .devcontainer/docker-compose.yml:23]: This is a good idea, but I've run into issues where I try to run two compose projects and find out I have port collisions. For other services, I've made this available but commented it out, which I feel like is a nice compromise: https://github.com/infratographer/permissions-api/blob/b5f6e6c34c22ba511270802a2a0dacef36ea00ff/.devcontainer/docker-compose.yml#L21-L22

Maybe not necessary here though. YMMV

Comment (jacobsee) [reply .devcontainer/docker-compose.yml:23]: I think... I would prefer this one to stay defaulted like this to be honest, just because I have run this devcontainer a lot in conjunction with the other projects I think somebody would be likely to have open at the same time (the governor UI, all the addons, permissions-api as of late etc) and those don't collide. The downside of defaulting it commented out like that would be requiring devs to catch one more setup thing in the readme and getting questions when they don't catch it... and we already get plenty of questions around correct env values for dev and such as-is.

tl;dr I think it's unlikely to be an issue here specifically and I'm afraid of adding more barriers to potential onboarding devs

**Review by jnschaeffer — APPROVED** [review]

**Review by fishnix — APPROVED** [review]
